lib/experimental/fonts-api/fonts-api.php

```diff
@@ -191,6 +191,11 @@ function wp_print_fonts( $handles = false ) {
 
 		_wp_scripts_maybe_doing_it_wrong( __FUNCTION__ );
 
+		// Automatically enqueue all user-selected fonts.
+		if ( false === $handles ) {
+			WP_Fonts_Resolver::enqueue_user_selected_fonts();
+		}
+
 		return $wp_fonts->do_items( $handles );
```

Comment:
I don't have a full understanding of how `WP_Fonts::do_items()` works, so my question might be a bit basic.
`WP_Fonts_Resolver::enqueue_user_selected_fonts();` returns a list of fonts, but that list is not being used anywhere.
Should it be passed to the `$wp_fonts->do_items()` method? Please see the example below to understand what I mean:
```suggestion
		// Automatically enqueue all user-selected fonts.
		if ( false === $handles ) {
			$auto_enqueued_fonts = WP_Fonts_Resolver::enqueue_user_selected_fonts();
		}

		return $wp_fonts->do_items( isset( $auto_enqueued_fonts ) ? $auto_enqueued_fonts : $handles );
```

Reply:
Good question @anton-vlasenko.

tl;dr
No, the handles do not need to be injected into `do_items()`.

Why?

Passing handles into `do_items()` only prints those handles and none of the ones already enqueued (i.e. in `WP_Fonts::$queue` property).

When `wp_print_fonts()` gets called, the goal is to print all enqueued fonts. These fonts will include:
- the theme's fonts (via `theme.json`), which are enqueued when the fonts get registered;
- the user-selected fonts (fonts from other sources than the theme's `theme.json`, e.g. plugins).

I designed `WP_Fonts_Resolver::enqueue_user_selected_fonts()` to return the font-families it enqueued in case it's needed in the future. It's currently being used in the automated tests. It's possible it could be needed in the source code at some point. Including it now means the method return signature won't need to change in the future (i.e. for BC).

What do you think @anton-vlasenko? Does that make sense?